KCardGrid part 2: Add KCardGrid with the base layouts + Add selection controls + Final documentation for KCard and KCardGrid + Few KCard bugfixes and improvements

MisRob opened the pull request:
## Description

### `KCard` bugfixes, visual and public API updates

- Adds support for selection controls such as checkboxes
-  Updates related to spaces between sections:
   - Aligns spaces more closely to designs
   - Introduces more robust spacing technique to ensures spaces are rendered as expected no matter of whether some/all/none slots are displayed
   - Improves calculations coordinating space taken by the thumbnail and other content in the horizontal orientation so that no free space is wasted (progressive update taking effect in browsers that support `clamp()`)
 
| Before | After |
| --------- | ------- |
| ![Screenshot from 2024-09-17 20-57-49](https://github.com/user-attachments/assets/47276508-18fd-4227-b24b-94945d9615d1) | ![Screenshot from 2024-09-17 20-58-57](https://github.com/user-attachments/assets/7b19cb06-e24f-4036-b201-7fbd66f60a8e) |

- Hides the placeholder after the thumbnail image is loaded
- Renames `titleLines` prop to `titleMaxLines`
- Renames `layout` prop to `orientation`

### `KCardGrid`

- Introduces `KCardGrid` with the base layouts
- `KCard` updates to make screen readers announce only card titles when navigating the grid with TAB key

### Docs

- Organizes and adds more guidance for `KCard` docs
- Adds new `KCardGrid` docs
- New documentation components to help with organizing larger amounts of guidance: `DocsSubNav`, `DocsToggleButton`, `DocsToggleContent`, `DocsTable`
- Slightly increases the width of the main documentation area to better demonstrate grids 

---

Also has some minor cleanup.

See details and reasons for changes in commit messages.

## References

- [Figma card grid designs](https://www.figma.com/design/lbzqOURWzAM5s21Jm3hEmi/Kolibri-Card-Breakpoints-%2F-Grids?node-id=1-506&t=sXBRNDkXulRaJRdC-0) - the three `KCardGrid` base layouts correspond to the (currently) only grid layouts we will use, and designs also demonstrate why grids are based on window breakpoints in contrast to available space in the containing element (however I think that may potentially be a valuable future improvement to be considered)
- Builds on https://github.com/learningequality/kolibri-design-system/pull/748
- Closes https://github.com/learningequality/kolibri-design-system/issues/264
- Closes https://github.com/learningequality/kolibri-design-system/issues/696
- Closes https://github.com/learningequality/kolibri-design-system/issues/697
- Closes https://github.com/learningequality/kolibri-design-system/issues/703

## Changelog

<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG START -->

  - **Description:** Renames `KCard`'s `titleLines` prop to `titleMaxLines`
  - **Products impact:** updated API
  - **Addresses:** -
  - **Components:** `KCard`
  - **Breaking:** yes
  - **Impacts a11y:** no
  - **Guidance:** Rename the prop

  - **Description:** Renames `KCard`'s `layout` prop to `orientation`
  - **Products impact:** updated API
  - **Addresses:** -
  - **Components:** `KCard`
  - **Breaking:** yes
  - **Impacts a11y:** no
  - **Guidance:** Rename the prop

  - **Description:** Adds support to `KCard` for selection controls such as checkboxes
  - **Products impact:** new API
  - **Addresses:** -
  - **Components:** `KCard`
  - **Breaking:** no
  - **Impacts a11y:** yes
  - **Guidance:** -

  - **Description:** Improves spaces display in `KCard`
  - **Products impact:** visual update
  - **Addresses:** -
  - **Components:** `KCard`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -
  
  - **Description:** Hides `KCard` placeholder element after the thumbnail image is loaded
  - **Products impact:** bugfix
  - **Addresses:** Resolves issue when parts of the placeholder element was visible behind a small thumbnail image 
  - **Components:** `KCard`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -

  - **Description:** Updates `KCard` to make screen readers announce only card titles when navigating the grid with TAB key
  - **Products impact:** bugfix
  - **Addresses:** Resolves unexpected behavior when overwhelming amounts of information was announced when navigating the card list with TAB 
  - **Components:** `KCard`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -
 
  - **Description:** Introduces `KCardGrid` (with internal related `KCard` updates)
  - **Products impact:** new API
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/697, https://github.com/learningequality/kolibri-design-system/issues/703
  - **Components:** `KCardGrid`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -

  - **Description:** Adds detailed guidance to `KCard` and `KCardGrid` documentation pages
  - **Products impact:** none
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/264, https://github.com/learningequality/kolibri-design-system/issues/696
  - **Components:** `KCard`, `KCardGrid`
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -
 
  - **Description:** Adds new documentation components: `DocsSubNav`, `DocsToggleButton`, `DocsToggleContent`, `DocsTable`
  - **Products impact:** none
  - **Addresses:** Helps with organizing larger amounts of guidance on documentation pages
  - **Components:** -
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -
 
  - **Description:** Slightly increases the width of the main documentation area
  - **Products impact:** none
  - **Addresses:** Helps to better demonstrate components requiring more space on documentation pages, such as card grids
  - **Components:** -
  - **Breaking:** no
  - **Impacts a11y:** no
  - **Guidance:** -

<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG END -->

## Steps to test

- Read both documentation pages for [KCard](https://deploy-preview-785--kolibri-design-system.netlify.app/kcard/) and [KCardGrid](https://deploy-preview-785--kolibri-design-system.netlify.app/kcardgrid/)
  - Is all guidance clear?
  - Is there some missing guidance?

- Test live examples on both documentation pages
  - Preview resulting markup
  - Try keyboard navigation
  - Try screen reader output 

I can also provide a copy of the playground page with all visual layouts. However now when there's many live examples on documentation pages, for the purpose of review it would make more sense to use them as they offer larger variety regarding card content.

## Testing checklist
<!-- Complete the checklist before submitting a PR; delete anything that doesn't apply -->

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [ ] The change is described in the changelog section above

## Reviewer guidance
<!-- Delete anything that doesn't apply so your reviewer knows what to check for -->

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

## Comments

- This PR completes all main pieces currently needed in Kolibri and Studio. After it's merged, we can close https://github.com/learningequality/kolibri-design-system/pull/748 and link to KDS latest release.
- Next PRs in the series will introduce `useKCardGrid` for advanced grid customization and loading states

MisRob committed:
Move playground to a directory

to have a suitable location
for playground sub-components
when testing complex scenarios

MisRob committed:
Add higher quality sample picture

alongside the existing very small picture
to allow for testing in both cases.

MisRob committed:
Add space below the belowTitle slot

to visually separate it from the footer slot.

MisRob committed:
Increase documentation content area size

to allow components that required more space
to demonstrate their function in relevant ways,
such as grids.

MisRob committed:
Fine-tune spacing and make it robust

Ensures spaces are rendered as expected
no matter of whether some/all/none slots
are displayed. Also aligns spaces more closely
to designs.

MisRob committed:
Fix unprecise calculation

and use constants to define thumbnail width
so they can be later referenced instead
of hardcoded values..

MisRob committed:
Coordinates space taken by the thumbnail area and

the content area next to it more intelligently
in browsers that support `clamp()` by:

 - Instead of defining 'width', 'min-width', and 'max-width' separately,
   `clamp()` is used with the goal to have the actual thumbnail width
   saved in the single `$thumbnail-width` value.

- The `$thumbnail-width` value is then referenced when calculating
  the remaining space for the content area, ensuring the precise
  distribution of space.

Resolves some issues related to unprecise calculations, most importantly
this removes the area of empty space between the thumbnail and content areas
in some card's sizes, wasting space that can be used for card's textual content.

MisRob committed:
Hide the placeholder after the thumbnail image is loaded

Otherwise in some scenarios, such as when there is a large
placeholder element and a small thumbnail image, some parts
of the placeholder element may be visible behind the image
after it has been successfully loaded.

MisRob committed:
Add KCardGrid with base layouts

MisRob committed:
Add new documentation components

MisRob committed:
Add new KCardGrid docs and improve KCard docs

MisRob committed:
Rename titleLines to titleMaxLines

to communicate its function more clearly

MisRob committed:
Rename layout to orientation

to communicate its purpose.

The card layout is rather a combination
of its orientation and the thumbnail area
configuration.

MisRob committed:
Use available constants

MisRob committed:
Add support for selection controls

MisRob committed:
Remove uneccessary doc folders

MisRob committed:
Fix component name

MisRob committed:
Fix screenreaders announcing all card content

when navigating the list of cards with TAB key.

This resolves overwhelming output and aligns the screenreader
navigation to the expected experience when on TAB only card titles
should be announced (other content is still accessible by  navigating
with arrows).

The main culprit that caused all content of cards being announced
was 'tabindex=0' set on <li>. To resolve the problem,
an alternative technique that doesn't require this tabindex is used
to make the focus ring display around the whole card area.

MisRob committed:
Fix bookmark button's purpose not being announced

by screenreaders and add some docs.

MisRob committed:
Move card components to one directory

MisRob committed:
Reset playground

MisRob committed:
Overview and Guidelines sections writing improvements

MisRob committed:
Allow the docs toggle button be displayed inline next to related text

MisRob committed:
Usage sections writing improvements

MisRob committed:
Remove duplicate information of API documentation sections

This information, with more, is now documented
in the main documentaton sections.

MisRob commented:
I'd like to ask

- @radinamatic for QA (see Steps to Test) and read through on both documentation pages. Now it's time for detailed QA on KDS side - feel free to test keyboard and screen reader experience for all live examples on the pages.
- @akolson @LianaHarris360 @AllanOXDi @AlexVelezLl @marcellamaki - would be grateful if you could read the two documentation pages
-  @AllanOXDi @AlexVelezLl - would you review code?

Note that even though the documentation build shows as failed in the PR checks, I was able to re-run it. Just use the links to the docs in the PR description. Thanks all :)

MisRob commented:
@radina Regarding checkboxes toggling working only with SPACE but not ENTER, I noticed ENTER is not working on our `KCheckboxes` in general. So if that was expected experience, we will need to fix, just not in the scope of this work. Will wait for your confirmation and can open issue.

MisRob commented:
I'd like to ask

- @radinamatic for QA (see Steps to Test) and read through on both documentation pages. Now it's time for detailed QA on KDS side - feel free to test keyboard and screen reader experience for all live examples on the pages.
- @jtamiace @akolson @LianaHarris360 @AllanOXDi @AlexVelezLl @marcellamaki - would be grateful if you could read the two documentation pages
-  @AllanOXDi @AlexVelezLl - would you review code?

~Note that even though the documentation build shows as failed in the PR checks, I was able to re-run it. Just use the links to the docs in the PR description.~ Fixed

Thanks all :)

AllanOXDi commented:
> @AllanOXDi @AlexVelezLl - would you review code?
I will review it. Thanks

MisRob committed:
Specify the direction of cards stretching in docs

MisRob committed:
Do not use links for API references on a same page

to prevent overwhelming visual otput.

MisRob committed:
Condense related info to the same paragraph

MisRob commented:
Thanks @AllanOXDi :)

For documentation pages code, I think there's no need to spend much time on it - it's large diff and if something went so wrong, we'd spot it on the generated docs. cc @AlexVelezLl  

I think in one of the upcoming PRs, I'd like to move live examples to sub-components to make further updates of documentation simpler. Perhaps something we can experiment with and discuss as a potential pattern.

MisRob commented:
Thanks @AllanOXDi :)

For documentation pages code, I think there's no need to spend much time on it - it's large diff and if something went so wrong, we'd spot it on the generated docs. cc @AlexVelezLl  

I think in one of the upcoming PRs, I'd like to move live examples to sub-components to make further updates of documentation simpler. Relatedly, I'd like to review @AlexVelezLl's idea about toggling code. Perhaps something we can begin experimenting with and discuss as a potential pattern. Let's chat in meetings :)

MisRob commented:
Thanks @AllanOXDi :)

For documentation pages code, I think there's no need to spend much time on it - it's large diff and if something went so wrong, we'd spot it on the generated docs. cc @AlexVelezLl  

I think in one of the upcoming PRs, I'd like to move live examples to sub-components to make further updates of documentation simpler. Relatedly, I'd like to revive @AlexVelezLl's idea about toggling code. Perhaps something we can begin experimenting with and discuss as a potential pattern. Let's chat in meetings :)

MisRob committed:
More realistict cards sample and add more content variety

AllanOXDi commented:
Hi @MisRob I started reviewing  this PR . Overall, the component is very robust and thoughtfully designed for flexibility and reusability.  I have posted  few  comments  for optimizations around code reuse, style handling, and event management. Thank you so much for your great work here.

AllanOXDi commented on lib/cards/KCard.vue:
```diff
@@ -177,34 +200,32 @@
-       * unique within a grid and descriptive to aid screenreader
-       * navigation.
+       * Card title
        */
```
Since title prop is set  to null by default. We  might want to add an additional check  to ensure that either the title prop or the title slot is always present since this is critical for a for KCard .

MisRob replied:
The check is in `KCard`'s `mounted`, looks good or do you think it needs to be adjusted somehow?

AllanOXDi replied:
You're right, the check in the mounted hook does ensure that either the title slot or prop is present. I was just thinking that since the title prop defaults to null, an explicit check for null or undefined might be worth considering, just to avoid any edge cases where the prop is technically passed but empty. Otherwise, it looks good!

MisRob replied:
@AllanOXDi Would you post steps to reproduce the issue you're describing? To understand what you mean I think I will need to see the actual code that is causing the check to not announce what it's supposed to.

MisRob replied:
@AllanOXDi this doesn't seem blocking and I've just addressed remaining feedback so we can merge before the next release - still you're welcome to let me know and I will take care of it in one of the upcoming PRs.

AllanOXDi commented on lib/cards/KCard.vue:
```diff
@@ -273,31 +288,58 @@
       },
-      layoutClass() {
-        return this.stylesAndClasses.layoutClass;
+      cardAreaClasses() {
```
To me it is  appearing that there are minor redundancies in ` cardAreaClasses` and `thumbnailStyles`. It might be worth refactoring these into helper functions to reduce code duplication, especially for the various combinations of orientation and thumbnail sizes?

MisRob replied:
I think I'd need to hear a bit more to understand. `thumbnailStyles` are passed as `appearanceOverrides` prop to `KImg`, which is quite different from what `cardAreaClasses` do, so I'm not entirely sure how to connect these. If you had ideas on how to simplify, that's always nice - would you write some pseudo-code to demonstrate how you imagined the helper functions?

AllanOXDi replied:
I was thinking that even though they serve different purposes, they can be simplified by focusing on the specific scenarios they handle  large or small thumbnail, horizontal or vertical orientation

```

function getCardAreaClasses(orientation) {
    // Return classes based on orientation
  

function getThumbnailStyles(size, additionalStyles = {}) {
    const baseStyles = {
        // Add  base styles here
    };

    return { ...baseStyles, ...additionalStyles };
}

```

To manages the visual aspect of the image based on the orientation (horizontal or vertical) and thumbnailSize (large or small). It returns an object with styles, simplifying logic related to spacing, sizing, and fitting the image.

AlexVelezLl replied:
I dont get it @AllanOXDi all of this is handled in the [stylesAndClasses](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L353) computed prop, and there we already manage the logic for large/small/horizontal/vertical layouts, and I dont see duplicated code there, and yes, we do have some baseStyles for both of them: [thumbnailCommonStyles](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L354), [cardAreaCommonClasses](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L358).

MisRob replied:
@AllanOXDi Let's look at

```
function getCardAreaClasses(orientation) {
// Return classes based on orientation
```

for example. This can't be achieved so easily, since `orientation` is not the only deciding factor (for one example, see horizontal orientation and larger thumbnail display). So in these new helpers, we'd basically need to start duplicating the orientation/thumbnailDisplay (=layout) conditions that are already in `stylesAndClasses`. Taking into account that main motivation for this particular organization was to avoid having the layout definitions scattered all over the place via many same conditions, I am afraid this strategy wouldn't help us overall. I can understand that it's a larger method and no problem to split it if you have other ideas, however it would need to happen in a way that would allow us to keep the single source of truth for layout behavior. Without it, it proved to be quite a pain point complicating any updates.

AllanOXDi commented on lib/cards/KCardGrid.vue:
```diff
@@ -0,0 +1,106 @@
+   */
+  export default {
+    name: 'KCardGrid',
+
+    setup(props) {
+      const { currentLevelConfig } = useResponsiveGridLayout(props);
+
+      const gridStyle = ref({});
+      const gridItemStyle = ref({});
+
+      watch(
+        currentLevelConfig,
+        newValue => {
+          const { cardsPerRow, columnGap, rowGap } = newValue;
+
+          gridStyle.value = {
+            'column-gap': columnGap,
+            'row-gap': rowGap,
+          };
```
It would be helpful to include more detailed comments here  to make it easier for others like me

MisRob replied:
I'm happy to. What areas would need clarification? E.g. what questions you had when reading.

AllanOXDi replied:
Thanks for offering to add more details! I just wanted to clarify how `currentLevelConfig` is determined by `useResponsiveGridLayou`t, the reasoning behind the calc function for` gridItemStyle` (especially how it handles gaps between cards).

MisRob replied:
There's already a comment for `gridItemStyle` few line below

```
// remove all column gaps widths from the available width and then divide by the number of cards per row to get a single card width
```

Is that clear enough?

MisRob replied:
There's already a comment for `gridItemStyle` few lines below

```
// remove all column gaps widths from the available width and then divide by the number of cards per row to get a single card width
```

Is that clear enough or what are areas that need some adjustment?

AllanOXDi replied:
I think it's clear enough

AllanOXDi commented on lib/cards/KCardGrid.vue:
```diff
@@ -0,0 +1,106 @@
+    name: 'KCardGrid',
+
+    setup(props) {
+      const { currentLevelConfig } = useResponsiveGridLayout(props);
```
I think  let's destructure props to make it clearer which properties are being used.

MisRob replied:
They're destructured below in the `watch` every time the value changes. Is that sufficient? If I descructured them sooner, we would need to watch every single value separatedly.

AllanOXDi replied:
That makes sense—I hadn't considered that destructuring too early would require separate watchers.

AllanOXDi commented on lib/cards/KCard.vue:
```diff
@@ -150,19 +167,25 @@
   export default {
     name: 'KCard',
+    setup() {
+      // provided by `KCardGrid`
```
I think we change this comments to `//gridItemStyle controls the width and layout of KCard items in the grid.` either from here on KCardGrid

MisRob replied:
That'd be nice. Done.

AllanOXDi commented on docs/pages/kcardgrid.vue:
```diff
@@ -0,0 +1,563 @@
+
+      <p>Grid's visual behavior is based on the <DocsInternalLink text="window breakpoint system" href="/layout#responsiveness" />. <code>KCardGrid</code> determines how many cards per row to display based on its layout and a current window breakpoint.</p>
+
+      <p><code>KCardGrid</code> doesn't manage inner card content. This is <code>KCardGrid</code>'s responsibility.</p>
```
Contradictory Statement here. It might be clearer to state that KCardGrid does not manage the content but only the layout of the cards.

MisRob replied:
Ah yes, good catch. It was supposed to say "This is KCard's responsibility." Fixed.

MisRob committed:
Comment on gridItemStyle

MisRob committed:
Fix docs typo

MisRob committed:
Remove redundant documentation

Documentation page is better suited for this and it
already explains grid layouts in detail.

MisRob commented:
Thanks a lot @AllanOXDi. I pushed some updates and left couple of clarifying questions.

AlexVelezLl commented on docs/pages/kcard.vue:
```diff
@@ -1,459 +1,732 @@
+        <KCardGrid ...>
+          <KCard ...>
+            <template #thumbnailPlaceholder>
+              <span :style="{ display: 'flex', height: '100%', justifyContent: 'center', alignItems: 'center' }">
```
I wonder if this `span` with its styles is something that we could have in the inner implementation of KCard, to make the public API a little bit easier to use. I think having a placeholder that doesnt have a height of 100% and its not centered would look a little weird (?) and adding this as a restriction of the inner implementation could have benefits in avoiding weird layouts, and also less boilerplate for the dev who is using KCard.

Also as this is something its recommended to always set, but mostly _just in case_ something goes wrong with the image, simplifying this could help in have a better addoption of this slot.

So we could have just

```Vue
<KCard ...>
  ...
  <template #thumbnailPlaceholder>
    <KIcon
      :style="{ fontSize: '48px' }"
      icon="readSolid"
    />
  </template>
</KCard>
```

MisRob replied:
Yes, that's useful. Done, thank you.

AlexVelezLl commented on docs/pages/kcardgrid.vue:
```diff
@@ -0,0 +1,563 @@
+      <p><em>Setting height on cards is discouraged. Instead, manage height bottom-up, for example by setting height on card sections, using text truncation, or other ways to limit its inner content.</em> Such approaches ensure content tolerance, prevent from unexpected overflows or excessive height, and keep vertical alignment of card sections consistent on a grid row. This is especially importa ...
+
+      <DocsShow language="html">
+        <KCardGrid layout="1-2-3">
```
Im not completely sure if this is part of the scope of this PR, but it seems that we have a small bug on responsiveness when we pass from 2 columns to just 1 column. I was wondering if this a KCardGrid thing or this was because of the docs not being responsive. But I saw that this empty space is controlled by KCardGrid so its the responsibility of KCardGrid.

https://github.com/user-attachments/assets/0cfdef90-fc40-4601-9346-308260ade057

MisRob replied:
Yes, seems to be a bug with grid behavior. I will investigate.

MisRob replied:
Done, it was likely caused by unpreciseness (perhaps some rounding errors) of `flex-basis`. Using `width` instead helped.

AlexVelezLl commented on lib/cards/KCardGrid.vue:
```diff
@@ -0,0 +1,97 @@
+
+      // provide to `KCard`
+      // controls the width and layout of `KCard` items in the grid
+      provide('gridItemStyle', gridItemStyle);
```
really loved this strategy to keep the public api cleaner

MisRob replied:
Thanks

AlexVelezLl reviewed (commented):
Thank you @MisRob!! This is looking great!! Loved how KCard and KCardGrid docs are structured, and code changes looks good to me. I left just a couple of questions/concerns, that Im not sure if falls in the scope of this PR. Thanks!

MisRob committed:
Always use annotation color

for text below title

MisRob committed:
Center placeholder content by default

MisRob committed:
More noticeable show more button

LianaHarris360 commented:
I've read over the documentation a bit and only noticed a few things:
1. There's a typo in this sentence

![kcard1](https://github.com/user-attachments/assets/f417df97-d233-4ad7-b900-477a20838e55)


2. When you click on a card within the documentation, you see an error message saying "This page could not be found." Could we make it so that the card is still clickable, but clicking it just keeps the user on the same page (or takes the user nowhere or perhaps to the top of that section) instead of opening that page? 

![kcard2](https://github.com/user-attachments/assets/55691fad-4d3e-4c1e-a5e0-6aa7a0a0a7b4)

radinamatic commented:
Excellent work @MisRob , I'm so 😍 with the progress here! 👏🏽 🏅 💯 

Even though some further attention will be necessary for proper labeling of the interactive elements within the card, this is a very strong and accessible foundation! Keyboard navigation is fluid, without hickups, and with the order we agreed upon, screen reader output is according to the expectations! 🎉 

Bookmark | Checkbox
-- | -- 
<img width="1280" alt="KCard" src="https://github.com/user-attachments/assets/573201db-cec0-4793-847c-75f5eb630a00"> | <img width="1280" alt="KCard2" src="https://github.com/user-attachments/assets/d818c9e9-236a-41d8-9103-2e6d6520aa17">

> Regarding checkboxes toggling working only with SPACE but not ENTER, I noticed ENTER is not working on our KCheckboxes in general. So if that was expected experience, we will need to fix, just not in the scope of this work. Will wait for your confirmation and can open issue.

According to the [W3C ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox/#kbd_label), the expected checkbox pattern requires only the `Space` key to change the state from checked to unchecked and back, so we should be good 🙂

radinamatic reviewed (approved):
Accessibility-wise, this is ready to 🚀! 🙂 🎉 :shipit:

marcellamaki commented:
@MisRob - I have read through the documentation and find it comprehensive and well-organized. There may be things that we end up wanting to add or adjust as we start implementing (and I think it would be ideal if some of the engineers who are less familiar with this do some of the refactoring work in 0.18, so we can get a new perspective on the work and documentation and see what additional information is useful for the team. (I haven't done any code review, as it seems as though that's been covered by others, and that Radina has done comprehensive accessibility QA as well)

MisRob committed:
Fix incorrect number of cards displayed

on a grid row by not using `flex-basis`
that seems to not be as precise as using
`width`.

MisRob committed:
Fix typo

MisRob committed:
Link doc cards to the same page

MisRob committed:
Align sample cards with designs

MisRob committed:
Improve doc live and code examples display

MisRob commented:
@LianaHarris360 Thank you.

(1) Fixed
(2) Yes, that'd be nice. I redirected to the "Guidelines" section so we can still easily recognize navigation works.

MisRob commented:
Good news @radinamatic, thank you

MisRob commented:
Thanks for reading through it @marcellamaki. Yes, it'd be great if cards could be now used by more developers who would give feedback.

MisRob commented:
All actionable feedback addressed. I am going to update changelog and open 'beta' -> 'this final version' migration issues for existing Kolibri and Studio cards and then we can merge.

MisRob committed:
Remove unused class

MisRob committed:
Reduce space between below title and footer

MisRob committed:
Consistent style for pills

MisRob committed:
Revert unwanted style change
